### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by Smiley73**
## Description

Initial release

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,62 @@
+BINARY = prometheus_bigquery_remote_storage_adapter
```
Comment (seanmalloy) [inline Makefile:1]: Wrong binary name here.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+.PHONY: build
+build:
+	CGO_ENABLED=0 go build -o $(BINARY) -ldflags $(LDFLAGS)
+
+.PHONY: image
+image:
+	docker build . -t quay.io/kohlstechnology/blackbox-helloworld-responder:latest
+
+.PHONY: test
+test: fmt vet test-unit
```
Comment (seanmalloy) [inline Makefile:26]: Should probably add `lint` to the list here too.

File: README.md [hunk 3]
```diff
@@ -1,39 +1,82 @@
...
+In order to build the docker image, simply execute
+
+```
+make image
+```
+
+## Testing
+
+### Unit Test
+`make test`
```
Comment (MeganMoran) [inline README.md:65]: wrap with triple backticks like other code sections for consistency

File: README.md [hunk 4]
```diff
@@ -1,39 +1,82 @@
...
+```
+
+How to manually create a release without relying on Travis CI.
+```
+$ TAG=v0.0.2 make tag
+$ GITHUB_TOKEN=xxx make clean release
+```
 
+### Test Release
+`make test-release`
```
Comment (MeganMoran) [inline README.md:82]: wrap with triple backticks like other code sections for consistency

File: README.md [hunk 5]
```diff
@@ -1,39 +1,82 @@
...
+
+### Unit Test
+`make test`
+
+## Releasing
+
+This project is using [goreleaser](https://goreleaser.com). GitHub release creation is automated using Travis
+CI. New releases are automatically created when new tags are pushed to the repo.
+```
+$ TAG=v0.0.2 make tag
```
Comment (MeganMoran) [inline README.md:72]: Assuming this is locked down permissions wise?

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+	LINT_INPUT="$(shell go list ./...)"; golint -set_exit_status $$LINT_INPUT
+
+.PHONY: vet
+vet:
+	VET_INPUT="$(shell go list ./...)"; go vet $$VET_INPUT
+
+.PHONY: tag
+tag:
+	git tag -a $(TAG) -m "Release $(TAG)"
+	git push origin $(TAG)
```
Comment (MeganMoran) [inline Makefile:57]: tag requires GITHUB_TOKEN environment variable to be set too right?

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,11 @@
+FROM gcr.io/distroless/static:nonroot
+
+COPY ./blackbox-helloworld-responder /blackbox-helloworld-responder
+
+ENV HELLO_WORLD_HTTP_PORT=8080
+ENV HELLO_WORLD_TCP_PORT=8081
+
+EXPOSE 8080/tcp
```
[the remark this answers is not in the document]
Comment (Smiley73) [reply Dockerfile:8]: Just to limit it to TCP only, to not expose UDP.

File: Makefile (hunk 6 above)
Comment (Smiley73) [reply Makefile:57]: No, we use tag locally to "release", so it uses your own credentials. See the readme file for the details.

File: README.md (hunk 5 above)
Comment (Smiley73) [reply README.md:72]: Yes you need push access to the repo

File: main.go [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+	"time"
+
+	"github.com/KohlsTechnology/blackbox-helloworld-responder/pkg/version"
+)
+
+func main() {
+	log.SetOutput(os.Stdout)
+	log.Println(version.Get())
+
+	var portHTTP int = 8080
```
[the remark this answers is not in the document]
Comment (Smiley73) [reply main.go:35]: It's the default setting. If the env variable isn't defined, then it uses those values.

**Pull request merged**